pkg/virt-controller/network/ipamclaims_test.go

```diff
@@ -0,0 +1,198 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/mock/gomock"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/api"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+
+	fakenetworkclient "kubevirt.io/client-go/generated/network-attachment-definition-client/clientset/versioned/fake"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+	fakeipamclaimclient "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1/apis/clientset/versioned/fake"
+)
+
+const (
+	redNetworkLogicalName  = "red"
+	redNetworkNadName      = redNetworkLogicalName + "-net"
+	blueNetworkLogicalName = "blue"
+	blueNetworkNadName     = blueNetworkLogicalName + "-net"
+	namespace              = "test-ns"
+	vmiName                = "testvmi"
+	vmiUID                 = vmiName + "UID"
+	nadNetworkName         = "nad_network_name"
+)
+
+var _ = Describe("CreateIPAMClaims", func() {
+	var virtClient *kubecli.MockKubevirtClient
+	var networkClient *fakenetworkclient.Clientset
+	var vmi *virtv1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		networkClient = fakenetworkclient.NewSimpleClientset()
+		virtClient.EXPECT().NetworkClient().Return(networkClient).AnyTimes()
+		ipamClaimsClient := fakeipamclaimclient.NewSimpleClientset()
+		virtClient.EXPECT().IPAMClaimsClient().Return(ipamClaimsClient).AnyTimes()
+	})
+
+	BeforeEach(func() {
+		vmi = api.NewMinimalVMI(vmiName)
+		vmi.Namespace = namespace
+		vmi.Spec.Domain.Devices.Interfaces = []virtv1.Interface{
+			{Name: redNetworkLogicalName},
+			{Name: blueNetworkLogicalName},
+			{Name: "absent", State: virtv1.InterfaceStateAbsent},
+		}
+		vmi.UID = vmiUID
+		vmi.Spec.Networks = []virtv1.Network{
+			logicalSecondaryNetwork(redNetworkLogicalName, redNetworkNadName),
+			logicalSecondaryNetwork(blueNetworkLogicalName, blueNetworkNadName),
+			logicalSecondaryNetwork("absent", "absent-net"),
+		}
+	})
+
+	Context("With allowPersistentIPs enabled in the NADs", func() {
+		BeforeEach(func() {
+			persistentIPs := map[string]bool{redNetworkNadName: true, blueNetworkNadName: true}
```

Comment:
nit: why a map string bool ? What does the boolean give you ? 

Why can't you just use a `map[string]struct{}' if you just check the presence of a key in the map ?